### PR TITLE
Add Kaldi compatibility test for `sliding_window_cmn`

### PR DESCRIPTION
**Pull request by mthrok**
Follow up of #540 

This PR adds
 - Kaldı compatibility test
 - ~~batch consistency test~~
for `sliding_window_cmn`

~~and make `sliding_window_cmn` handle batched input.~~

cc @wanglong001 

EDIT: batch ompatibility part was moved to #570

### CODE REVIEW

File: test/test_batch_consistency.py [hunk 1]
```diff
@@ -69,6 +69,13 @@ def test_contrast(self):
         waveform = torch.rand(2, 100) - 0.5
         _test_batch(F.contrast, waveform, enhancement_amount=80.)
 
+    def test_sliding_window_cmn(self):
+        waveform = torch.randn(2, 1024) - 0.5
+        _test_batch(F.sliding_window_cmn, waveform, center=True, norm_vars=True)
+        _test_batch(F.sliding_window_cmn, waveform, center=True, norm_vars=False)
+        _test_batch(F.sliding_window_cmn, waveform, center=False, norm_vars=True)
+        _test_batch(F.sliding_window_cmn, waveform, center=False, norm_vars=False)
```
Comment (mthrok) [inline test/test_batch_consistency.py:77]: We should definitely use `pytest.mark.parametrize` to simplify this.

**Comment by mthrok** [comment]
The changes I made in `functional.py` are necessary to pass the test, but I am not sure if the tests cover all the branches in the function. Adding coverage report to test would help finding out this kind of edge case.

File: test/test_batch_consistency.py (hunk 1 above)
Comment (vincentqb) [reply test/test_batch_consistency.py:76]: You can try `subTest` in `unittest`.

Comment (mthrok) [reply test/test_batch_consistency.py:76]: Our test runner does not understand `subTest`. The test name does not show up in the execution result, so `subTest` does not add any value despite more code and indentation.

File: test/test_kaldi_compatibility.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    args = []
+    for key, value in kwargs.items():
+        key = '--' + key.replace('_', '-')
+        value = str(value).lower() if value in [True, False] else str(value)
+        args.append('%s=%s' % (key, value))
+    return args
+
+
+def _run_kaldi(command, input_tensor):
+    """Run provided Kaldi command, pass a tensor and get the resulting tensor
```
Comment (vincentqb) [inline test/test_kaldi_compatibility.py:25]: Since kaldi is not installed in CI, the test is skipped in travis
```
(test/test_kaldi_compatibility.py::TestFunctional::test_sliding_window_cmn SKIPPED [100%]
```

For the other tests, we simply saved the output from kaldi, and compare against that. This allows us to run the test on CI. Let's do that.

If you want to keep the infrastructure you have added to run kaldi commands, we can do it too, along with the test just mentioned. We can rename this as e.g. `test_sliding_window_cmn_against_installed_kaldi`.

Comment (mthrok) [reply test/test_kaldi_compatibility.py:25]: My concern with the previous approach is that it adds overhead for both adding a new test and maintaining the test, which leads to the situation where every time a new Kaldi-compatible functionality is added, one of us has to add all the related files, because it is a complicated process to review.

When we run tests on Circle CI, we can bake Kaldi binaries into base test image. Making Kaldi accessible on CI should be easy.

Comment (vincentqb) [reply test/test_kaldi_compatibility.py:25]: I do also prefer using kaldi commands to test against for new transforms, over testing against saved kaldi output files. This has the advantage of making sure we stay in sync with kaldi, and we can regenerate the exact command that produces the kaldi output we want.

Adding kaldi is easy (though there is always a risk that it won't run easily on some platform like windows), but we should isolate this change in a separate PR from the testing. And, since kaldi is not yet available in CircleCI, this test is not running and so is not showing us that `sliding_window_cmn` is compatible with kaldi.

Even if kaldi were available, we do not have convincing evidence in this PR's CircleCI that the new test is indeed providing the signal we want.

I see two options:
* We add kaldi in the testing build now.
* We add a second test against the kaldi output file, along with what you have here.

Let's go for the latter in this PR, and follow up with the former.

**Comment by mthrok** [comment]
@vincentqb 

I used the new image being added in https://github.com/pytorch/audio/pull/581 to run the Kaldi compatibility test and it passed.

Search `test_kaldi_compatibility` in `Run test` step.

https://app.circleci.com/pipelines/github/pytorch/audio/1514/workflows/e2be1e0d-641b-46e9-ae16-1775e3fed72d

**Comment by mthrok** [comment]
> * We could decide to switch all the kaldi tests to this.

I think we can create `help-wanted` tasks for that. I think it is really nice if we can get rid of the previous test and helper scripts.

> * We could start thinking of migrating tests from travis to circleci only once all tests on travis also run on circleci.

Sure.

**Comment by vincentqb** [comment]
> I think we can create `help-wanted` tasks for that. I think it is really nice if we can get rid of the previous test and helper scripts.

Good idea: #597
